--- GWorkspace-0.6.5/Desktop/GNUmakefile.preamble.orig	2004-05-17 13:01:00.000000000 +0200
+++ GWorkspace-0.6.5/Desktop/GNUmakefile.preamble	2004-07-05 23:07:54.856326656 +0200
@@ -21,6 +21,7 @@
 #                      -L../GWLib/$(GNUSTEP_OBJ_DIR) \
 #                      -L../../GWLib/$(GNUSTEP_OBJ_DIR)   
 #endif
+ADDITIONAL_LIB_DIRS += -L../FSNode/FSNode.framework/Versions/Current/$(GNUSTEP_TARGET_LDIR)
 
 ADDITIONAL_TOOL_LIBS +=
 
--- GWorkspace-0.6.5/Finder/GNUmakefile.preamble.orig	2004-05-29 12:37:27.000000000 +0200
+++ GWorkspace-0.6.5/Finder/GNUmakefile.preamble	2004-07-05 23:06:35.299401216 +0200
@@ -15,7 +15,7 @@
 ADDITIONAL_LDFLAGS +=  
 
 # Additional library directories the linker should search
-ADDITIONAL_LIB_DIRS += -L../FSNode/$(GNUSTEP_OBJ_DIR) 
+ADDITIONAL_LIB_DIRS += -L../FSNode/FSNode.framework/Versions/Current/$(GNUSTEP_TARGET_LDIR)
 
 ADDITIONAL_TOOL_LIBS +=
 
