diff -ur orig-GWorkspace-0.6.5/ClipBook/main.m GWorkspace-0.6.5/ClipBook/main.m
--- orig-GWorkspace-0.6.5/ClipBook/main.m	2003-10-28 05:46:19.000000000 -0700
+++ GWorkspace-0.6.5/ClipBook/main.m	2004-11-09 02:50:21.897824882 -0700
@@ -33,8 +33,15 @@
 int main(int argc, char **argv, char **env)
 {
 	CREATE_AUTORELEASE_POOL (pool);
-  ClipBook *clipbook = [ClipBook clipbook];
-	NSApplication *app = [NSApplication sharedApplication];
+  ClipBook *clipbook;
+	NSApplication *app;
+
+#ifdef GS_PASS_ARGUMENTS
+  [NSProcessInfo initializeWithArguments: argv count: argc environment: env];
+#endif
+ 
+	clipbook	= [ClipBook clipbook];
+	app	= [NSApplication sharedApplication];
   
 	createMenu();
 	
diff -ur orig-GWorkspace-0.6.5/Desktop/main.m GWorkspace-0.6.5/Desktop/main.m
--- orig-GWorkspace-0.6.5/Desktop/main.m	2004-05-26 06:07:30.000000000 -0600
+++ GWorkspace-0.6.5/Desktop/main.m	2004-11-09 02:50:21.897824882 -0700
@@ -32,8 +32,15 @@
 int main(int argc, char **argv, char **env)
 {
 	CREATE_AUTORELEASE_POOL (pool);
-  Desktop *desktop = [Desktop desktop];
-	NSApplication *app = [NSApplication sharedApplication];
+  Desktop *desktop;
+	NSApplication *app;
+
+#ifdef GS_PASS_ARGUMENTS
+  [NSProcessInfo initializeWithArguments: argv count: argc environment: env];
+#endif
+
+	desktop = [Desktop desktop];
+	app = [NSApplication sharedApplication];
 
 	createMenu();
 
diff -ur orig-GWorkspace-0.6.5/Finder/main.m GWorkspace-0.6.5/Finder/main.m
--- orig-GWorkspace-0.6.5/Finder/main.m	2004-04-13 17:23:41.000000000 -0600
+++ GWorkspace-0.6.5/Finder/main.m	2004-11-09 02:50:21.898824742 -0700
@@ -35,8 +35,15 @@
 int main(int argc, char **argv, char **env)
 {
 	CREATE_AUTORELEASE_POOL (pool);
-  Finder *finder = [Finder finder];
-	NSApplication *app = [NSApplication sharedApplication];
+  Finder *finder;
+	NSApplication *app;
+
+#ifdef GS_PASS_ARGUMENTS
+  [NSProcessInfo initializeWithArguments: argv count: argc environment: env];
+#endif
+
+	finder = [Finder finder];
+	app = [NSApplication sharedApplication];
   
 #ifdef GNUSTEP
 	createMenu();
diff -ur orig-GWorkspace-0.6.5/GWNet/main.m GWorkspace-0.6.5/GWNet/main.m
--- orig-GWorkspace-0.6.5/GWNet/main.m	2004-01-20 06:21:31.000000000 -0700
+++ GWorkspace-0.6.5/GWNet/main.m	2004-11-09 02:50:21.898824742 -0700
@@ -35,8 +35,15 @@
 int main(int argc, char **argv, char **env)
 {
 	CREATE_AUTORELEASE_POOL (pool);
-  GWNet *gwnet = [GWNet gwnet];
-	NSApplication *app = [NSApplication sharedApplication];
+  GWNet *gwnet;
+	NSApplication *app;
+
+#ifdef GS_PASS_ARGUMENTS
+  [NSProcessInfo initializeWithArguments: argv count: argc environment: env];
+#endif
+	
+	gwnet = [GWNet gwnet];
+	app = [NSApplication sharedApplication];
   
 #ifdef GNUSTEP
 	createMenu();
diff -ur orig-GWorkspace-0.6.5/GWorkspace/main.m GWorkspace-0.6.5/GWorkspace/main.m
--- orig-GWorkspace-0.6.5/GWorkspace/main.m	2004-05-17 04:42:29.000000000 -0600
+++ GWorkspace-0.6.5/GWorkspace/main.m	2004-11-09 02:50:21.898824742 -0700
@@ -38,8 +38,15 @@
 int main(int argc, char **argv, char **env)
 {
 	CREATE_AUTORELEASE_POOL (pool);
-  GWorkspace *gw = [GWorkspace gworkspace];
-	NSApplication *app = [NSApplication sharedApplication];
+  GWorkspace *gw;
+	NSApplication *app;
+
+#ifdef GS_PASS_ARGUMENTS
+  [NSProcessInfo initializeWithArguments: argv count: argc environment: env];
+#endif
+
+	gw	= [GWorkspace gworkspace];
+	app = [NSApplication sharedApplication];
   
 #ifdef GNUSTEP
 	createMenu();
diff -ur orig-GWorkspace-0.6.5/Inspector/main.m GWorkspace-0.6.5/Inspector/main.m
--- orig-GWorkspace-0.6.5/Inspector/main.m	2004-02-19 18:03:49.000000000 -0700
+++ GWorkspace-0.6.5/Inspector/main.m	2004-11-09 02:50:21.898824742 -0700
@@ -35,8 +35,15 @@
 int main(int argc, char **argv, char **env)
 {
 	CREATE_AUTORELEASE_POOL (pool);
-  Inspector *inspector = [Inspector inspector];
-	NSApplication *app = [NSApplication sharedApplication];
+  Inspector *inspector;
+	NSApplication *app;
+
+#ifdef GS_PASS_ARGUMENTS
+  [NSProcessInfo initializeWithArguments: argv count: argc environment: env];
+#endif
+
+	inspector = [Inspector inspector];
+	app	= [NSApplication sharedApplication];
   
 #ifdef GNUSTEP
 	createMenu();
diff -ur orig-GWorkspace-0.6.5/Operation/main.m GWorkspace-0.6.5/Operation/main.m
--- orig-GWorkspace-0.6.5/Operation/main.m	2004-03-08 06:09:17.000000000 -0700
+++ GWorkspace-0.6.5/Operation/main.m	2004-11-09 02:50:21.899824603 -0700
@@ -35,8 +35,15 @@
 int main(int argc, char **argv, char **env)
 {
 	CREATE_AUTORELEASE_POOL (pool);
-  Operation *operation = [Operation operation];
-	NSApplication *app = [NSApplication sharedApplication];
+  Operation *operation;
+	NSApplication *app;
+
+#ifdef GS_PASS_ARGUMENTS
+  [NSProcessInfo initializeWithArguments: argv count: argc environment: env];
+#endif
+
+	operation	= [Operation operation];
+	app	= [NSApplication sharedApplication];
   
 #ifdef GNUSTEP
 	createMenu();
diff -ur orig-GWorkspace-0.6.5/Utilities/findfile/findfile.m GWorkspace-0.6.5/Utilities/findfile/findfile.m
--- orig-GWorkspace-0.6.5/Utilities/findfile/findfile.m	2003-09-26 04:25:38.000000000 -0600
+++ GWorkspace-0.6.5/Utilities/findfile/findfile.m	2004-11-09 02:54:24.510959690 -0700
@@ -429,11 +429,16 @@
 
 @end
 
-int main(int argc, char** argv)
+int main(int argc, char** argv, char** env)
 {
 	FindFile *findfile;
   
   CREATE_AUTORELEASE_POOL (pool);
+
+#ifdef GS_PASS_ARGUMENTS
+  [NSProcessInfo initializeWithArguments: argv count: argc environment: env];
+#endif
+
 	findfile = [[FindFile alloc] init];
   
   if (findfile) {
diff -ur orig-GWorkspace-0.6.5/Utilities/fswatcher/fswatcher.m GWorkspace-0.6.5/Utilities/fswatcher/fswatcher.m
--- orig-GWorkspace-0.6.5/Utilities/fswatcher/fswatcher.m	2004-04-14 04:33:21.000000000 -0600
+++ GWorkspace-0.6.5/Utilities/fswatcher/fswatcher.m	2004-11-09 02:54:08.381211163 -0700
@@ -625,7 +625,7 @@
 @end
 
 
-int main(int argc, char** argv)
+int main(int argc, char** argv, char** env)
 {
 	FSWatcher *fsw;
 
@@ -641,6 +641,10 @@
 	  default:
 	    exit(0);
 	}
+
+#ifdef GS_PASS_ARGUMENTS
+  [NSProcessInfo initializeWithArguments: argv count: argc environment: env];
+#endif
   
   CREATE_AUTORELEASE_POOL (pool);
 	fsw = [[FSWatcher alloc] init];
diff -ur orig-GWorkspace-0.6.5/Utilities/thumbnailer/main.m GWorkspace-0.6.5/Utilities/thumbnailer/main.m
--- orig-GWorkspace-0.6.5/Utilities/thumbnailer/main.m	2003-09-26 04:25:38.000000000 -0600
+++ GWorkspace-0.6.5/Utilities/thumbnailer/main.m	2004-11-09 02:54:42.041512685 -0700
@@ -679,6 +679,10 @@
 	  default:
 	    exit(0);
 	}
+
+#ifdef GS_PASS_ARGUMENTS
+  [NSProcessInfo initializeWithArguments: argv count: argc environment: env_c];
+#endif
   
   pool = [NSAutoreleasePool new];
 	thumbnailer = [[Thumbnailer alloc] init];
diff -ur orig-GWorkspace-0.6.5/Utilities/wopen/wopen.m GWorkspace-0.6.5/Utilities/wopen/wopen.m
--- orig-GWorkspace-0.6.5/Utilities/wopen/wopen.m	2003-07-12 07:42:09.000000000 -0600
+++ GWorkspace-0.6.5/Utilities/wopen/wopen.m	2004-11-09 02:54:56.042558347 -0700
@@ -37,6 +37,10 @@
   NSString *fullPath = nil;
   BOOL isDir = NO;
   id gworkspace = nil;
+
+#ifdef GS_PASS_ARGUMENTS
+  [NSProcessInfo initializeWithArguments: argv count: argc environment: env_c];
+#endif
    
   pool = [NSAutoreleasePool new];
   fm = [NSFileManager defaultManager];
